src/iteration.rs

```diff
@@ -0,0 +1,295 @@
+use std::{
+    ffi::{CStr, CString},
+    io::{Read, Seek, SeekFrom, Write},
+    ops::DerefMut,
+    slice,
+};
+
+use libc::{c_int, c_void};
+
+use crate::{error::archive_result, ffi, ffi::UTF8LocaleGuard, Error, Result, READER_BUFFER_SIZE};
+
+struct HeapReadSeekerPipe<R: Read + Seek> {
+    reader: R,
+    buffer: [u8; READER_BUFFER_SIZE],
+}
+
+/// The contents of a file, yielded in order from the beginning to the end of
+/// the archive.
+///
+/// Each file, including directories, will have a StartOfFile, zero or more
+/// chunks of data, and then a corresponding EndOfFile to mark that the file is
+/// done.
+pub enum ArchiveContents {
+    /// Marks the start of a file, including for directories.
+    StartOfFile(String),
```

Comment:
```suggestion
    StartOfEntry(String),
```
This avoids the file word use, so it is more generic.